Add send/receive to `spdm_common_lib`

PrithviAPai opened the pull request:
Once secure session is established, in application phase both SPDM Requester and SPDM Responder can send/receive messages. Added patch which will enable SPDM Responders also to send message during Appication phase.

Resolves issue: https://github.com/DMTF/libspdm/issues/2571

jyao1 commented:
@PrithviAPai , please squash your commit.
There is no need to create a patch to fix format issue.

PrithviAPai commented:
> @PrithviAPai , please squash your commit. There is no need to create a patch to fix format issue.

Addressed the comment. 
Squashed the commits. Thank You

Wenxing-hou commented:
Thanks for your commit. And I compare the change: no code changes, only positional changes. LGTM.
Please submit commit too in spdm-emu if the PR is merged.

jyao1 commented:
Would you please show an example on how responder uses those APIs?

PrithviAPai commented:
> Would you please show an example on how responder uses those APIs?

SPDM Responder(Integrator) daemon can expose APIs for applications to send encrypted data to SPDM EndPoint.
The payload received by the applications can be sent to EID using `libspdm_send_data`
